app/src/main/java/com/github/steroidteam/todolist/user/User.java

```diff
@@ -0,0 +1,54 @@
+package com.github.steroidteam.todolist.user;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+
+public interface User {
+    /* Trivial method to have here */
+    boolean isLoggedIn();
+
+    /* Unique user identifier */
+    String getUserId() throws UserLoginException;
+
+    /* User profile */
+    String getDisplayName() throws UserLoginException;
+
+    String getEmail() throws UserLoginException;
+
+    boolean isEmailVerified() throws UserLoginException;
+
+    /* Actions on current user */
+    void updateDisplayName(String newDisplayName, OnCompleteListener<Void> onComplete)
+            throws UserLoginException;
+
+    void updateEmail(String newEmail, OnCompleteListener<Void> onComplete)
+            throws UserLoginException;
+
+    void sendEmailVerification(OnCompleteListener<Void> onComplete) throws UserLoginException;
+
+    void setUserPassword(String newPassword, OnCompleteListener<Void> onComplete)
+            throws UserLoginException;
+
+    void sendPasswordResetEmail(OnCompleteListener<Void> onComplete) throws UserLoginException;
+
+    void deleteUser(OnCompleteListener<Void> onComplete) throws UserLoginException;
+
+    void reauthenticate(OnCompleteListener<Void> onComplete);
```

Comment:
I am not a big fan of callbacks; personally I think that returning `CompletableFuture`s is better here.

In any case it is functional and not a major issue, just a matter of being as consistent as possible with the asynchronous code. So I would be okay with leaving this as-is if it is a big hassle.

Reply:
You're right. I added these callbacks by following the way the Firebase authentication's API works.